src/renderer/components/ft-video-player/ft-video-player.js

```diff
@@ -712,10 +721,21 @@ export default defineComponent({
     },
 
     updateVolume: function (_event) {
-      // 0 means muted
       // https://docs.videojs.com/html5#volume
-      const volume = this.player.muted() ? 0 : this.player.volume()
-      sessionStorage.setItem('volume', volume)
+      if (sessionStorage.getItem('muted') === 'false' && this.player.volume() === 0) {
+        // If video was muted by dragging volume slider, we reset the volume to default volume.
+        // dragging volume to 0 doesn't change muted in sessionStorage to true hence checking it with 'false'
+        const volume = parseFloat(sessionStorage.getItem('defaultVolume'))
+        const muted = true
+        sessionStorage.setItem('volume', volume)
+        sessionStorage.setItem('muted', muted)
+      } else {
+        // If video was muted by pressing 'm'/'M' or clicking on mute button, then we reset the volume to previous volume.
```

Comment:
This comment should probably be removed or changed to be accurate.
the `volumechange` event is fired whenever the volume or the muted state of the player fires, this comment makes it seem like this bit of code is only run when you mute the player, which is not true
it will run in situations like this one below as well
before:
muted: false
volume: 0.25
_volumechange event fires_
muted: false
volume: 0.3